unitTests/ui/SimpleFilterTest.ts

```diff
@@ -1,227 +1,327 @@
+        });
+
+        it('should reset checkboxes when clearBreadcrumb is triggered', () => {
+          aSimpleFilter.cmp.selectValue('foo');
+          aSimpleFilter.cmp.selectValue('bar');
+          let args: IPopulateBreadcrumbEventArgs = {
+            breadcrumbs: []
+          };
+          $$(aSimpleFilter.env.root).trigger(BreadcrumbEvents.populateBreadcrumb, args);
+          $$(aSimpleFilter.env.root).trigger(BreadcrumbEvents.clearBreadcrumb, args);
+          expect(aSimpleFilter.cmp.getSelectedCaptions().length).toEqual(0);
+        });
+
+        it('should not trigger a query when clearBreadcumb is triggered', () => {
+          $$(aSimpleFilter.env.root).trigger(BreadcrumbEvents.clearBreadcrumb);
+          expect(aSimpleFilter.env.queryController.executeQuery).not.toHaveBeenCalled();
+        });
+
+        it('should not log an analytics event when clearBreadcrumb is triggered', () => {
+          $$(aSimpleFilter.env.root).trigger(BreadcrumbEvents.clearBreadcrumb);
+          expect(aSimpleFilter.env.usageAnalytics.logSearchEvent).not.toHaveBeenCalled();
+        });
+
+        it('should redraw checkbox containers with the previously selected Values if they are present', () => {
+          aSimpleFilter.cmp.options.values = undefined;
+          aSimpleFilter.cmp.closeContainer();
+          const results = FakeResults.createFakeResults();
+          results.groupByResults = [FakeResults.createFakeGroupByResult('@field', 'foo', 5)];
+          Simulate.query(aSimpleFilter.env, {
+            results: results
+          });
+          aSimpleFilter.cmp.selectValue('foo1');
+          Simulate.query(aSimpleFilter.env);
+          expect(aSimpleFilter.cmp.getSelectedCaptions()[0]).toEqual('foo1');
+        });
+
+        it('should not render a clear button if enableClearButton is falsy', () => {
+          expect($$(aSimpleFilter.cmp.element).findAll('.coveo-simplefilter-eraser').length).toBe(
+            aSimpleFilter.cmp.options.enableClearButton ? 1 : 0
+          );
+        });
       });
 
-      aSimpleFilter.cmp.toggleContainer();
-    });
-
-    afterEach(() => {
-      aSimpleFilter = null;
-    });
-
-    it('should set the title correctly', () => {
-      expect(aSimpleFilter.cmp.options.title).toBe('FooTitleBar');
-    });
-
-    it('should set the field correctly', () => {
-      expect(aSimpleFilter.cmp.options.field).toBe('@field');
-    });
-
-    it('should allow to getValueCaption', () => {
-      aSimpleFilter.cmp.options.field = '@filetype';
-      expect(aSimpleFilter.cmp.getValueCaption('bar')).toBe('baz');
-    });
-
-    it('should set the values correctly', () => {
-      expect(aSimpleFilter.cmp.options.values).toEqual(['foo', 'bar']);
-    });
-
-    it('should set the sortCriteria correctly', () => {
-      expect(aSimpleFilter.cmp.options.sortCriteria).toBe('score');
+      describe(`SimpleFilter with more than one instance and enabledClearButton as ${enableClearButton}`, () => {
```

Comment:
Same kind of comment here, with that changes it makes it so it will be hard to modify that test file further to test other variation.

Right now it is structured a bit like if the `enableClearButton` was somehow a more "important" option in the component than all the others and that we have to run all unit tests with a variation on that option, and that does not seem right.